spec/unit/configurer_spec.rb

```diff
@@ -589,18 +589,36 @@ def method_missing(*args)
       expect(Puppet.settings.setting(:lastrunfile)).to receive(:mode).and_return('664')
       @configurer.save_last_run_summary(@report)
 
+      # Copy/Pasta from file_system_spec to see if we can get this working.
+      SYSTEM_SID_BYTES = [1, 1, 0, 0, 0, 0, 0, 5, 18, 0, 0, 0]
+
+      def is_current_user_system?
+        SYSTEM_SID_BYTES == Puppet::Util::Windows::ADSI::User.current_user_sid.sid_bytes
+      end
+
       if Puppet::Util::Platform.windows?
         require 'puppet/util/windows/security'
         mode = Puppet::Util::Windows::Security.get_mode(Puppet[:lastrunfile])
+        # Using a solution here similar to that employed for file_system_spec.rb to mitigate
+        # an issue running under appveyor
+        expected_perm = is_current_user_system? ? 0664 : 0644
+        expect(mode & 0777).to eq(expected_perm)
       else
         mode = Puppet::FileSystem.stat(Puppet[:lastrunfile]).mode
+        expect(mode & 0777).to eq(0664)
       end
-      expect(mode & 0777).to eq(0664)
     end
 
-    it "should report invalid last run file permissions" do
+    it "should report invalid last run file permissions with invalid mode", :if => Puppet.features.microsoft_windows? do
```

Comment:
This should not be filtered out because with the change in https://github.com/puppetlabs/puppet/pull/7594/files#diff-3bdc5fd6acc30b485dc30ba606c026d6R413 the old functionality is changed.

```
[1] pry(#<Puppet configuration client>)> Puppet.settings.setting(:lastrunfile).mode
=> "33333"
[2] pry(#<Puppet configuration client>)> Integer(Puppet.settings.setting(:lastrunfile).mode,8)
=> 14043
[3] pry(#<Puppet configuration client>)> Puppet::FileSystem.replace_file(Puppet[:lastrunfile], 14043) do |fh|
[3] pry(#<Puppet configuration client>)*   fh.print YAML.dump(report.raw_summary)
[3] pry(#<Puppet configuration client>)* end
=> 0
[4] pry(#<Puppet configuration client>)>
```